ruby_on_rails/apis/working_with_external_apis.md

```diff
@@ -31,10 +31,14 @@ You'll typically also get a "secret key" or similarly named code.  Whereas the A
 
 Most APIs require a different type of "security clearance" for different requests:  
 
+1. You can usually make innocuous requests like asking for posts from X's (formerly known as Twitter) API with straightforward and unauthenticated GET requests.  You can make these from any command line or a browser tool like [Postman](http://getpostman.com). These types of API requests are usually severely rate limited to incentivize you to sign up.  
+1. The next layer is making requests that include your API key. These are still usually fairly innocuous things (like getting public data) and limited by the API's pricing tiers.
+1. More sensitive requests like asking for specific user data or submitting/modifying/deleting data will likely require you to use an authentication process involving your secret token.  We'll cover the basics of this in the project.  Rates for these requests are subject to the API pricing tiers.
+1. Oftentimes, you actually want to make requests on behalf of a user.  For instance, showing a user a dashboard of all their posts and Facebook posts together would require asking X for that user's posts and Facebook for that user's posts.  This can involve a LOT of requests over a large user base, but luckily you're actually able to make them on the user's behalf by asking for the user's permission.  We'll cover this as well later, but basically you send the user to the API provider's site to sign in, then the API provider will give you a user-specific token to use when making requests on their behalf in the future.  Rates for this are usually more advantageous because they are covered in a per-user bucket.  We typically use the OAuth protocol for this, as described below.
```

Comment:
Looks like when you resolved the merge conflict, you accidentally accepted both original and incoming text, so both versions made it in :stuck_out_tongue: We can just removed these four points since they're duplicates (except for "browser tool" where the other says "application")
```suggestion
```